core/src/contracts/stablex_contract.rs

```diff
@@ -324,6 +323,27 @@ impl StableXContract for StableXContractImpl {
             .stream()
             .boxed()
     }
+
+    fn noop_transaction<'a>(&'a self, gas_price: U256) -> BoxFuture<'a, Result<TransactionResult>> {
```

Comment:
I'm not a huge fan of `cancel_pending_transaction`, especially since there are no parameter relating to a transaction. I think once we include gnosis/ethcontract-rs#345 then `fn cancel_pending_transaction(self, tx: TransactionResult)` makes a lot more sense, since we would be able to recover the tx nonce and gas price to send a proper transaction cancellation.

Until then `send_noop_transaction` feels better for me. Although, I like using an action verb like `send` to indicate its actually affecting the EVM.